### PR TITLE
dkg: add nil checks

### PR DESCRIPTION
**Pull request by dB2510**
Adds nil pointer checks for relevant functions in dkg package.

Refer: OBOL-03

category: misc
ticket: #1997

### CODE REVIEW

File: dkg/frostp2p.go [hunk 1]
```diff
@@ -32,7 +32,11 @@ var (
 )
 
 // newFrostP2P returns a p2p frost transport implementation.
-func newFrostP2P(tcpNode host.Host, peers map[peer.ID]cluster.NodeIdx, secret *k1.PrivateKey, threshold int) *frostP2P {
+func newFrostP2P(tcpNode host.Host, peers map[peer.ID]cluster.NodeIdx, secret *k1.PrivateKey, threshold int) (*frostP2P, error) {
+	if secret == nil {
+		return nil, errors.New("secret cannot be nil")
+	}
```
Comment (corverroos) [inline dkg/frostp2p.go:38]: I'm not worried about this, as the code should ensure not to pass in nil.

File: dkg/frostp2p.go [hunk 2]
```diff
@@ -122,6 +126,10 @@ func newFrostP2P(tcpNode host.Host, peers map[peer.ID]cluster.NodeIdx, secret *k
 	p2p.RegisterHandler("frost", tcpNode, round1P2PID,
 		func() proto.Message { return new(pb.FrostRound1P2P) },
 		func(ctx context.Context, pID peer.ID, req proto.Message) (proto.Message, bool, error) {
+			if req == nil {
```
Comment (corverroos) [inline dkg/frostp2p.go:129]: same here, the contract of `RegisterHandler` is that it will never pass in nil

File: dkg/frostp2p.go (hunk 1 above)
Comment (dB2510) [reply dkg/frostp2p.go:38]: cool, I'll revert this one

File: dkg/frostp2p.go (hunk 2 above)
Comment (dB2510) [reply dkg/frostp2p.go:129]: will revert this too

**Review by corverroos — APPROVED** [review]

**Pull request merged**